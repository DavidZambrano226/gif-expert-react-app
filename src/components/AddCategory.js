import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handlerChange = (e) => {
        setInputValue(e.target.value)
    }

    const handlerSubmit = (e) => {
        e.preventDefault();
        
        if (inputValue.trim().length > 2 ) {
            setCategories( category => [inputValue, ...category] );
            setInputValue('');
        }

    }
    
    return (
        <form onSubmit={handlerSubmit}>
            <input 
                type="text" 
                value = {inputValue}
                onChange={ handlerChange } 
                
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
