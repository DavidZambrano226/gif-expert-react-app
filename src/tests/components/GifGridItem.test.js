import { shallow } from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem"


describe('GifGridItem test component', () => {

    const imgData = {
        id: '1',
        title: 'Custom title test',
        url: 'http://someurl.test.com'
    }
    const wrapper = shallow(<GifGridItem {...imgData} /> )

    test('should make match with snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('should render the title in a p element in HTML', () => {
        const p = wrapper.find('p').text().trim();
        expect( p ).toBe( imgData.title );
    })

    test('should render the img with src and alt assigned', () => {
        const img = wrapper.find('img').props();
        expect( img.src ).toBe( imgData.url );
        expect( img.alt ).toBe( imgData.title );

    })
    
    test('should have a div container with class animate__zoomIn', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__zoomIn')).toBeTruthy();
    })
    
    
    

})
