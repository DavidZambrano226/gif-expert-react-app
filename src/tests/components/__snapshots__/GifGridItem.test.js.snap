// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem test component should make match with snapshot 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Custom title test"
    src="http://someurl.test.com"
  />
  <p>
    Custom title test
  </p>
</div>
`;
